servlet/src/main/java/io/grpc/servlet/ServerBuilder.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2018 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.servlet;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.grpc.Server;
+import io.grpc.ServerStreamTracer.Factory;
+import io.grpc.Status;
+import io.grpc.internal.AbstractServerImplBuilder;
+import io.grpc.internal.Channelz.SocketStats;
+import io.grpc.internal.GrpcUtil;
+import io.grpc.internal.Instrumented;
+import io.grpc.internal.InternalServer;
+import io.grpc.internal.LogId;
+import io.grpc.internal.ServerImpl;
+import io.grpc.internal.ServerListener;
+import io.grpc.internal.ServerTransport;
+import io.grpc.internal.ServerTransportListener;
+import io.grpc.internal.SharedResourceHolder;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Builder to build a gRPC server that can run as a servlet.
+ */
+@NotThreadSafe
+public final class ServerBuilder extends AbstractServerImplBuilder<ServerBuilder> {
```

Comment:
ServletServerBuilder? Let's not have it the same name as its ancestor.

Reply:
Done.